### PR TITLE
Debug Duplicate Glosses

### PR DESCRIPTION
**Pull request by KamBustos**
I am working on a bug of duplicate glosses appearing for this sprint. There are some Glosses that appear more than once. They are the exact same gloss -- same title and same @id. Bug will be fixed if they only appear once. You can view this bug at this link: https://glossing.rerum.io/glosses.html?gog-filter=JTdCJTIydGl0bGUlMjIlM0ElMjJOb24lMjB0ZXJyZWFudCUyMHJ1aW5lJTIyJTJDJTIydGV4dCUyMiUzQSUyMk5vbiUyMHRlcnJlYW50JTIwcnVpbmUlMjIlMkMlMjJ0YXJnZXRlZHRleHQlMjIlM0ElMjJOb24lMjB0ZXJyZWFudCUyMHJ1aW5lJTIyJTdE

Bug:
issue_89

I am currently working in the files of glosses.html and ng.js. The DockerFile is for testing and will be removed when finalized PR is requested for review.

### CODE REVIEW

File: glosses.html [hunk 1]
```diff
@@ -54,6 +54,39 @@
             setTimeout(() => location.href = "", 3000)
         })
 
+        function removeDuplicateDeerIds() {
```
Comment (thehabes) [inline glosses.html:57]: You do not need any of the code on this page.  As am I testing locally, I deleted all of this and it functions just fine.  Delete this from the PR.

File: js/ng.js [hunk 2]
```diff
@@ -92,20 +92,18 @@ addEventListener('deer-form-rendered', event => {
  */ 
 addEventListener('deer-updated', event => {
     const $elem = event.target
-    //Only care about witness form
+    // Only care about witness form
```
Comment (thehabes) [inline js/ng.js:95]: Not sure what go you here.  None of this code should be edited.  Remove all of these changes from the PR.

File: js/deer-config.js [hunk 3]
```diff
@@ -156,18 +156,25 @@ export default {
...
+            function removeDuplicates(list, uniqueProp) {
+                return list.filter((item, index, self) =>
+                    index === self.findIndex((t) => (
+                        t[uniqueProp] === item[uniqueProp]
+                    ))
+                );
+            }
             // Grab the cached expanded entities from localStorage.  Note that there is nothing to check on "staleness"
             const cachedFilterableEntities = localStorage.getItem("expandedEntities") ? new Map(Object.entries(JSON.parse(localStorage.getItem("expandedEntities")))) : new Map()
+            const deduplicatedList = removeDuplicates(obj[options.list], '@id')
```
Comment (thehabes) [inline js/deer-config.js:168]: You should be doing this inside `if(options.list)`.  If there is no `options.list` here, which is possible, you will get a null pointer exception.

File: js/deer-config.js [hunk 4]
```diff
@@ -156,18 +156,25 @@ export default {
                 <p class="filterNotice is-hidden"> Gloss filter detected.  Please note that Glosses will appear as they are fully loaded. </p>
                 <div class="totalsProgress" count="0"> {loaded} out of {total} loaded (0%).  This may take a few minutes.  You may click to select any Gloss loaded already.</div>
             </div>`
-            
+            function removeDuplicates(list, uniqueProp) {
+                return list.filter((item, index, self) =>
+                    index === self.findIndex((t) => (
+                        t[uniqueProp] === item[uniqueProp]
+                    ))
+                );
```
Comment (thehabes) [inline js/deer-config.js:164]: No semicolon.

File: js/deer-config.js [hunk 5]
```diff
@@ -156,18 +156,25 @@ export default {
...
+            }
             // Grab the cached expanded entities from localStorage.  Note that there is nothing to check on "staleness"
             const cachedFilterableEntities = localStorage.getItem("expandedEntities") ? new Map(Object.entries(JSON.parse(localStorage.getItem("expandedEntities")))) : new Map()
+            const deduplicatedList = removeDuplicates(obj[options.list], '@id')
+            const total = deduplicatedList.length
```
Comment (thehabes) [inline js/deer-config.js:169]: If `total` is dependent upon `deduplicatedList`, you will have to set it inside of `if(options.list)` as well.

**Review by thehabes — CHANGES_REQUESTED** [review]
I ran into an error when partially through this review.  The comments are applicable still, you have some changes to make and clean up to do.

http://localhost:8080/gloss-transcription.html?tpen-project=https://t-pen.org/TPEN/project/7084 does not load due to an error from the template.  Please investigate.

**Review by AbhilashKotha — APPROVED** [review]

**Review by thehabes — APPROVED** [review]

**Review by AbhilashKotha — APPROVED** [review]

**Pull request merged**